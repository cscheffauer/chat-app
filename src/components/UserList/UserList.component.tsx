import React from 'react';
import UserListEntry from '../UserListEntry/UserListEntry.component';

import './UserList.scss';

interface Props {
	userList: Object;
}

const UserList = ({ userList }: Props) => {
	return (
		<div className='userlist'>
			{Object.entries(userList).map(([key, user]) => (
				<UserListEntry key={key} username={user.username} />
			))}
		</div>
	);
};

export default UserList;
