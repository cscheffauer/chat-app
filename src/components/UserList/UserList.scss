@import '../../utils/variables';

.userlist {
	height: 84vh; /* Fallback for browsers that do not support Custom Properties */
	height: calc(var(--vh, 1vh) * 84);
	background-color: $color-chat-window;
	border-left: $border-style-tabs;
	border-right: $border-style-tabs;
}
