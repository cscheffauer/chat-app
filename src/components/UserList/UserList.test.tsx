import React from 'react';
import { shallow } from 'enzyme';
import UserList from './UserList.component';

it('expect to render UserList component', () => {
    const userListWrapper = shallow(<UserList userList={{ abcd: { username: 'Max Mustermann' } }} />);
    expect(userListWrapper).toMatchSnapshot();
})
