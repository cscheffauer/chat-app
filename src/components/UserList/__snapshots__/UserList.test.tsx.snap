// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render UserList component 1`] = `ShallowWrapper {}`;
